client/js/_author/reducers/items.js

```diff
@@ -1,42 +1,56 @@
-import _          from 'lodash';
-import guid       from '../../utils/guid';
-import genusTypes from '../../constants/genus_types';
+import _                       from 'lodash';
+import guid                    from '../../utils/guid';
+import { types, getQbankType } from '../../constants/genus_types';
 
 // Leave this empty. It will hold assessments by bank id. IE `state[someId] = {a_bank}`
 const initialState = {};
 
-
-function updateChoiceData(item) {
-  const newItem = _.cloneDeep(item);
-
-  if (!newItem.question) {
-    newItem.question = {
-      choices: {},
-    };
-    return newItem;
-  }
-
+function deserializeChoices(choices, answers) {
   const newChoices = {};
-  _.forEach(item.question.choices, (choice, index) => {
+
+  _.forEach(choices, (choice, index) => {
     newChoices[choice.id] = {
+      id: choice.id,
+      answerId: null,
+      text: choice.text,
       order: index,
-      ...choice,
-      correct: false
+      feedback: null,
+      fileIds: [],
+      isCorrect: false,
     };
-    _.forEach(item.answers, (answer) => {
+    _.forEach(answers, (answer) => {
       if (_.includes(answer.choiceIds, choice.id)) {
         newChoices[choice.id] = {
-          answer,
           ...newChoices[choice.id],
           feedback: _.get(answer, 'feedback.text'),
-          correct: answer.genusTypeId === genusTypes.answer.rightAnswer,
+          isCorrect: answer.genusTypeId === types.answer.rightAnswer,
           answerId: answer.id,
         };
       }
     });
   });
-  newItem.question.choices = newChoices;
-  return newItem;
+  return newChoices;
+}
+
+// The implementation of function is Q-Bank specific
+function deserializeItem(item) {
```

Comment:
A lot of this seems very multiple choice specific. How does this handle things that don't have choices?